src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -282,26 +386,39 @@ private int GetNextLogNumberForToday()
             }
         }
 
-        private void CleanupOldLogFiles()
+        private void CleanupOldLogFiles(string[] logFiles)
         {
-            string[] existingLogFiles;
-            try
-            {
-                // we use GetFiles because we don't expect to have a massive amount of files, and it is 
-                // not sure what kind of iteration order we get if we run and modify using Enumerate
-                existingLogFiles = Directory.GetFiles(_path, "*.log");
-            }
-            catch (Exception)
+            var now = DateTime.Now;
+            foreach (var logFile in logFiles)
             {
-                return; // this can fail for various reasons, we don't really care for that in most cases
+                var logDateTime = File.GetLastWriteTime(logFile);
```

Comment:
I do not like this. We were extracting a date from file name here, for me it was better approach than rely on the last write time.